src/containerapp/azext_containerapp/_params.py

```diff
@@ -120,6 +120,7 @@ def load_arguments(self, _):
         c.argument('traffic_weights', nargs='*', options_list=['--traffic-weight'], help="A list of revision weight(s) for the container app. Space-separated values in 'revision_name=weight' format. For latest revision, use 'latest=weight'")
         c.argument('workload_profile_name', options_list=['--workload-profile-name', '-w'], help="Name of the workload profile to run the app on.", is_preview=True)
         c.argument('secret_volume_mount', help="Path to mount all secrets e.g. mnt/secrets", is_preview=True)
+        c.argument('termination_grace_period', type=int, options_list=['--termination-grace-period', '--grace-period'], help="Duration in seconds a replica is given to gracefully shut down before it is forcefully terminated. (Default: 30)", is_preview=True)
```

Comment:
is it support in --yaml?

Reply:
The termination_grace_period already existed in  _sdk_models, so it is supported in --yaml.
@njuCZ  Could you please add some test for property termination_grace_period in test_containerapp_create_with_yaml or add new test case?

Reply:
only api-version >= 2023-04-01-preview support this property. we are still using python sdk 2022-11-01-preview, so it's not supported in --yaml

Reply:
you can manual add this property

Reply:
updated